src/domTree.js

```diff
@@ -43,13 +43,9 @@ interface HtmlDomInterface extends VirtualNodeInterface {
     tryCombine(sibling: HtmlDomNode): boolean;
 }
 
-/**
- * All `HtmlDomNode`s must implement HtmlDomInterface.
- *
- * `HtmlDomNode` is not defined as an interface since `documentFragment` also
- * has these fields but should not be considered a `HtmlDomNode`.
- */
-export type HtmlDomNode = DomSpan | SvgSpan | anchor | symbolNode;
+/** All `HtmlDomNode`s must implement HtmlDomInterface. */
+export type HtmlDomNode =
+    DomSpan | SvgSpan | anchor | symbolNode | documentFragment;
```

Comment:
If all `HtmlDomNode`s implement `HtmlDomInterface`, couldn't we use `HtmlDomInterface` in place of `HtmlDomNode`?

Reply:
I'd considered that previously and now that you mention it again, I'm reconsidering my decision.

To elaborate on my original intention: the type sum is technically stricter and also allows one to clearly see which discrete possibilities apply (e.g. `span<SomethingElse>` is not allowed even through it implements the interface). My original concern was that `interface`s are strictly structural -- even if you don't explicitly say `implements HtmlDominterface`, something can accidentally implement it and allow us to pass in a value of that type.

However, given that `HtmlDomInterface` currently includes `tryCombine` and we have explicit constructors for `DomSpan` and `SvgSpan`, there's pretty much no chance of something accidentally satisfying the interface. So you're right, we could probably just replace it with the interface now.

Done. We can also revert it if ever needed as the change is pretty trivial.